tests/python/support/test_auto_weight.py

```diff
@@ -85,42 +86,19 @@ def test_detect_weight_same_dir_config_json(weight_format, index_filename, resul
         config_json_path = base_path / "config.json"
         _create_json_file(config_json_path, {})
         if index_filename is not None:
-            weight_index_file = os.path.join(tmpdir, index_filename)
+            weight_index_file = Path(os.path.join(tmpdir, index_filename))
             _create_json_file(weight_index_file, {})
-        assert detect_weight(None, config_json_path, weight_format) == (base_path, result)
+        assert detect_weight(None, config_json_path, weight_format) == (weight_index_file, result)
 
 
 def test_find_weight_fail():
     with tempfile.TemporaryDirectory() as tmpdir:
         base_path = Path(tmpdir)
         with pytest.raises(ValueError):
-            detect_weight(Path("do/not/exist"), base_path, "AWQ")
+            detect_weight(Path("do/not/exist"), base_path, "awq")
     with pytest.raises(AssertionError):
-        detect_weight(None, Path("do/not/exist"), "AWQ")
+        detect_weight(None, Path("do/not/exist"), "awq")
 
 
 if __name__ == "__main__":
-    test_detect_weight("PyTorch", "pytorch_model.bin.index.json", "PyTorch")
```

Comment:
no need to revert this change - the reason we wanted to keep it tedious is that we want the test to be runnable out of box with no extra steps to understand any internals of the system